components/Flashcard/Flashcard.js

```diff
@@ -32,6 +39,41 @@ export default function Flashcard({
     level,
   } = flashcard;
 
+  const motionValue = useMotionValue(0);
+  const rotateValue = useTransform(motionValue, [-200, 200], [-50, 50]);
+  const opacityValue = useTransform(
+    motionValue,
+    [-200, -150, 0, 150, 200],
+    [0, 1, 1, 1, 0]
+  );
+  const animControls = useAnimation();
+
+  function handleDragEnd() {
+    (event, info) => {
+      const { offset } = info;
+
+      if (!showAnswer) return;
+
+      if (offset.x > 150) {
+        onSwipe("right");
+        animControls.start({ x: 400, opacity: 0 }).then(() => {
+          motionValue.set(0);
+          animControls.start({ x: 0, opacity: 1 });
+        });
+      } else if (offset.x < -50) {
+        onSwipe("left");
+        animControls.start({ x: -400, opacity: 0 }).then(() => {
+          motionValue.set(0);
+          animControls.start({ x: 0, opacity: 1 });
+        });
+      } else {
+        animControls.start({ x: 0, opacity: 1 });
+      }
+    };
+  }
```

Comment:
I think this function also needs a little work. 

We can get the distance of the drag from the `motionValue` object.
We no longer need to worry about resetting the animation if you set the value of the dragConstraints to 0. 
(This also means we can remove the final else). 
I optionally put in that the card should be re-flipped at the end of the drag event (but you can decide if you want this or not, or when it should happen. 
Maybe something like:

``` js
  function handleDragEnd() {
    if (!showAnswer) return;
    const x = motionValue.get();

    console.log(x);

    if (x > 30) {
      console.log("increase");
      onSwipe("right");
      onIncreaseFlashcardLevel(id);
    } else if (x < -30) {
      console.log("decrease");
      onSwipe("left");
      onDecreaseFlashcardLevel(id);
    }
    setShowAnswer(false);
  }

```